src/app/services/toast.service.ts

```
@@ -0,0 +1,165 @@
+        closeButton.addEventListener('click', () => this.removeToast(toastId));
+      }
+    }
+
+    // if there's an existing toast with the same ID, remove before adding the new one
+    const existingToast = this.toasts.get(toastId);
+    if (existingToast) {
+      const { element, timeoutId } = existingToast;
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+      if (element && element.parentNode) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)